django/db/models/query.py

```diff
@@ -1719,3 +1721,36 @@ def get_related_populators(klass_info, select, db):
         rel_cls = RelatedPopulator(rel_klass_info, select, db)
         iterators.append(rel_cls)
     return iterators
+
+
+class FilteredRelation(object):
+    def __init__(self, relation_name, alias, condition):
+        if not relation_name:
+            raise exceptions.FieldError("FilterRelation expects a non-empty relation_name")
```

Comment:
Generally we don't repeat the name of the class/function in which the error occurred in the message, this information is already in the traceback. Also I'm not sure `FieldError` is much better than `ValueError` here

I suggest "relation_name cannot be empty" as the message